File: <100/035/035.jl
using Primes


function rotate(n:: Int):: Int
  ord = n |> string |> length |> (x -> x-1)
  lsd = n % 10
  n   = n ÷ 10
  n   = n + lsd * 10^ord
  return n
end


function rotates(n:: Int):: Array{Int}
  arr = Array{Int}([])
  ord = n |> string |> length
  for _ ∈ 1:ord
    n = rotate(n)
    push!(arr, n)
  end
  return arr
end


function all_rotates_primes(n:: Int):: Bool
  rotations = rotates(n)
  bools     = map(isprime, rotations)
  return bools |> all
end


function main():: Nothing
  upper_limit :: Int = 10^6
  ps    = primes(upper_limit)
  bools = map(all_rotates_primes, ps)
  bools |> count |> println

  return nothing
end


@time main()

File: <100/035/035.py
import sympy
import time


def rotate(n: int) -> int:
    ord = len(str(n)) - 1
    lsd = n % 10
    n   = n // 10 + lsd * 10**ord

    return n


def rotates(n: int) -> list[int]:
    ord       = len(str(n))
    rotations = [0 for _ in range(ord)]
    for j in range(ord):
        n = rotate(n)
        rotations[j] = n
    
    return rotations


def all_rotations_primes(n: int) -> bool:
    rotations = rotates(n)
    bools = map(sympy.isprime, rotations)
    return all(bools)


def main() -> int:
    upper_limit = 10**6
    primes = sympy.primerange(upper_limit)
    bools  = map(all_rotations_primes, primes)
    answer = sum(bools)
    return answer


if __name__ == "__main__":
    start = time.time()
    answer = main()
    elapsed = time.time() - start

    print(f'{elapsed} sec')
    print(answer)
